### PR TITLE
Board side panel

### PR DESCRIPTION
**Pull request by sanbornsen**
<strong>Note: If there are pending changes to the PR, prefix the PR title with "WIP" and add the label "DO NOT MERGE"</strong>

This PR adds the side panel to the board view. There are some behavioral changes in the side panel.

1. If the the board view is opened and any type group is selected then the navigation will stay inside the board view.

2. In any view, any navigation other than type group will take to the respective list page. 

3. Board pages will not have any filter momentarily and there is no filter expression option for the URL as well. 

4. Navigate to the board page by clicking on the second level header navigation or by directly hitting plan/board

### Mandatory
 - [ ] What does this PR do? 

- [ ] What issue/task does this PR references?

- [ ] Are the tests Included?
___
### Optional
- [ ] Is the documentation Included?

- [ ] Are the Release Notes included?

- [ ] @mention(s) to expected reviewer(s) included

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @sanbornsen for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


Your plugin configuration is stored in the [file](https://github.com/fabric8-ui/fabric8-planner/blob/3d7061750b63df96eaef5949dcdd9c02016b3c34/.ike-prow/test-keeper.yaml).

File: src/app/components_ngrx/group-types-panel/group-types-panel.component.ts [hunk 1]
```diff
@@ -27,6 +24,7 @@ import { AppState } from './../../states/app.state';
 export class GroupTypesComponent implements OnInit, OnDestroy {
 
   @Input() sidePanelOpen: boolean = true;
+  @Input() context: string; // 'list' or 'board'
```
Comment (vikram-raj) [inline src/app/components_ngrx/group-types-panel/group-types-panel.component.ts]: Can you make it as ``` @Input() context: 'list' | 'board' ```?

File: src/app/components_ngrx/iteration-list-entry/iteration-list-entry.component.ts [hunk 2]
```diff
@@ -43,6 +43,7 @@ export class IterationListEntryComponent implements OnInit, OnDestroy {
   @Input() witGroup: string = '';
   @Input() showTree: string = '';
   @Input() showCompleted: string = '';
+  @Input() context: string; // 'list' or 'board'
```
Comment (vikram-raj) [inline src/app/components_ngrx/iteration-list-entry/iteration-list-entry.component.ts]: Same here make it as ``` @Input() context: 'list' | 'board'````?

File: src/app/components_ngrx/iteration-tree/iteration-tree.component.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ export class IterationTreeComponent {
   @Input() witGroup: string = '';
   @Input() showTree: string = '';
   @Input() showCompleted: string = '';
+  @Input() context: string; // 'list' or 'board'
```
Comment (vikram-raj) [inline src/app/components_ngrx/iteration-tree/iteration-tree.component.ts]: Same here make it as ``` @Input() context: 'list' | 'board' ```?

File: src/app/components_ngrx/iterations-panel/iterations-panel.component.ts [hunk 4]
```diff
@@ -40,6 +39,7 @@ export class IterationComponent implements OnInit, OnDestroy, OnChanges {
   @Input() showTree: string = '';
   @Input() showCompleted: string = '';
   @Input() infotipText: string = '';
+  @Input() context: string; // 'list' or 'board'
```
Comment (vikram-raj) [inline src/app/components_ngrx/iterations-panel/iterations-panel.component.ts]: Same here make it as ``` @Input() context: 'list' | 'board'````?

File: src/app/components_ngrx/planner-board/planner-board.component.ts [hunk 5]
```diff
@@ -1,10 +1,109 @@
...
+      private route: ActivatedRoute,
+      private router: Router
+    ) {}
+
+    ngOnInit() {
+      this.iterationQuery.deselectAllIteration();
+      this.eventListeners.push(
+        this.spaceQuery.getCurrentSpace
+          .do(() => {
+            // this.uiLockedSidebar = true;
```
Comment (vikram-raj) [inline src/app/components_ngrx/planner-board/planner-board.component.ts]: Don't we need this?

File: src/app/components_ngrx/side-panel/side-panel.component.ts [hunk 6]
```diff
@@ -17,15 +8,12 @@ import { AppState } from './../../states/app.state';
 export class SidepanelComponent implements OnInit {
 
   @Input() sidePanelOpen: boolean = true;
+  @Input() context: string = 'list'; // 'list' or 'board'
```
Comment (vikram-raj) [inline src/app/components_ngrx/side-panel/side-panel.component.ts]: Same here make it as ``` @Input() context: 'list' | 'board'````?

File: src/app/components_ngrx/custom-query-panel/custom-query-panel.component.ts [hunk 7]
```diff
@@ -31,6 +26,7 @@ import { AppState } from './../../states/app.state';
 export class CustomQueryComponent implements OnInit, OnDestroy {
 
   @Input() sidePanelOpen: boolean = true;
+  @Input() context: string; // 'list' or 'board'
```
Comment (vikram-raj) [inline src/app/components_ngrx/custom-query-panel/custom-query-panel.component.ts]: Same here make it as ``` @Input() context: 'list' | 'board'````?

File: src/app/components_ngrx/planner-board/planner-board.component.ts (hunk 5 above)
Comment (sanbornsen) [reply src/app/components_ngrx/planner-board/planner-board.component.ts]: Not at this moment.. We can live without  it.

Comment (vikram-raj) [reply src/app/components_ngrx/planner-board/planner-board.component.ts]: Then can you remove this do block?

File: src/app/components_ngrx/group-types-panel/group-types-panel.component.ts [hunk 8]
```diff
@@ -1,21 +1,18 @@
-import { cloneDeep } from 'lodash';
 import { Observable } from 'rxjs/Observable';
-import { Subscription } from 'rxjs/Subscription';
 
-import { Component, Input, OnChanges, OnDestroy, OnInit } from '@angular/core';
-import { ActivatedRoute, Params, Router } from '@angular/router';
```
Comment (vikram-raj) [inline src/app/components_ngrx/group-types-panel/group-types-panel.component.ts]: We are using Router in this component. Please add it back.

File: src/app/components_ngrx/custom-query-panel/custom-query-panel.component.less [hunk 9]
```diff
@@ -25,57 +22,81 @@
...
+    .dropdown-kebab-pf {
+      .btn-link {
+        color: @color-pf-white;
+      }
+      .dropdown-menu-right {
+        right: -10px;
+      }
     }
   }
 }
```
Comment (vikram-raj) [inline src/app/components_ngrx/custom-query-panel/custom-query-panel.component.less]: @sanbornsen You did something wrong with this file while resolving the conflict. @SMahil Can you please look into it.

Comment (SMahil) [reply src/app/components_ngrx/custom-query-panel/custom-query-panel.component.less]: sure

File: src/app/components_ngrx/group-types-panel/group-types-panel.component.ts (hunk 8 above)
Comment (vikram-raj) [reply src/app/components_ngrx/group-types-panel/group-types-panel.component.ts]: Added Router in https://github.com/fabric8-ui/fabric8-planner/pull/2710/commits/cc0a547d05bed1ca96c22ba02816f66d677062df

File: src/app/components_ngrx/custom-query-panel/custom-query-panel.component.less (hunk 9 above)
Comment (vikram-raj) [reply src/app/components_ngrx/custom-query-panel/custom-query-panel.component.less]: CSS is fixed in https://github.com/fabric8-ui/fabric8-planner/pull/2710/commits/b17540c1fb013eb90084ac85654b06b686fc724f. Thanks @SMahil

**Pull request merged**